### PR TITLE
adding write STORE to .zip plus some evaluator rewiring

### PR DESCRIPTION
**Pull request by curtiskoo**
This PR adds the following:
- write to zip from string using https://www.archiverjs.com/docs/quickstart
- rewiring evaluating `tick` trigger to also create an advancement
- write load function tag into `minecraft` name space for `load` trigger (this is done a bit weirdly as we get the namespace from top level; feel free to refactor this)

### CODE REVIEW

File: src/codeGenerator/evaluator.ts [hunk 1]
```diff
@@ -434,22 +434,29 @@ export class Evaluator implements ExpressionVisitor {
...
+            advName = this.storeAdvancementFromTrigger(advName, fnName, astNode, env);
         }
         this.updateStore(fnName, fnValue, astNode);
         return advName;
     }
 
+    private storeAdvancementFromTrigger(advName: string, fnName: string, astNode: On, env: EvaluatorEnv) {
+        advName = `.adv${fnName}`;
+        let triggers = this.parseTrigger(astNode.trigger, env);
+        let advValue = new Store.AdvancementValue(
```
Comment (kilowatt-) [inline src/codeGenerator/evaluator.ts:453]: Best practices suggest that these should be `const` but I'm not going to insist on this change; just FYI

**Review by kilowatt- — APPROVED** [review]

File: src/codeGenerator/evaluator.ts (hunk 1 above)
Comment (curtiskoo) [reply src/codeGenerator/evaluator.ts]: i was lazy and just did the `refactor > extract` without checking, but addressed, thanks

**Review by JustinHuPrime — APPROVED** [review]

File: src/codeGenerator/triggerEvaluator.ts [hunk 2]
```diff
@@ -22,9 +22,8 @@ export default class TriggerEvaluator implements TriggerVisitor {
     visitLoad(astNode: Load) {
         throw new DSLSyntaxError(astNode, "load and tick triggers cannot be combined");
```
Comment (jlu5) [inline src/codeGenerator/triggerEvaluator.ts:23]: This should probably just say `load` now.

File: src/codeGenerator/store.ts [hunk 3]
```diff
@@ -19,19 +29,23 @@ export class FunctionValue implements Serializable, Writeable {
   }
 
   static onTick(name: string, commands: string[]) {
```
Comment (jlu5) [inline src/codeGenerator/store.ts:31]: If we're rewriting tick handling to use an advancement, I don't think we'll need this to be a separate static constructor anymore?

File: src/codeGenerator/triggerEvaluator.ts (hunk 2 above)
Comment (curtiskoo) [reply src/codeGenerator/triggerEvaluator.ts:23]: ack

File: src/codeGenerator/store.ts (hunk 3 above)
Comment (curtiskoo) [reply src/codeGenerator/store.ts:31]: ack

**Review by JustinHuPrime — APPROVED** [review]

**Review by jlu5 — APPROVED** [review]

**Pull request merged**